controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -398,32 +404,61 @@ public void createScope(ScopeInfo request, StreamObserver<CreateScopeStatus> res
     @Override
     public void listStreamsInScope(Controller.StreamsInScopeRequest request, StreamObserver<Controller.StreamsInScopeResponse> responseObserver) {
         String scopeName = request.getScope().getScope();
-        RequestTag requestTag = requestTracker.initializeAndTrackRequestTag(requestIdGenerator.get(), "listStream", scopeName);
+        RequestTag requestTag = requestTracker.initializeAndTrackRequestTag(requestIdGenerator.get(),
+                "listStream", scopeName);
         log.info(requestTag.getRequestId(), "listStream called for scope {}.", scopeName);
+
+        final AuthContext ctx;
+        if (this.grpcAuthHelper.isAuthEnabled()) {
+            ctx = AuthContext.current();
```

Comment:
will AuthContext.current return the last authContext?
What if there were multiple interceptors. Will there be multiple authContexts?
If so, this AuthContext.current will not retrieve all the contexts, right?

Reply:
The logic here works in the presence of multiple auth interceptors. Let me explain this in detail. 

`AuthContext.current()` will return an object containing the `Principal` and `AuthHandler` objects. The `Principal` is directly retrieved from the current gRPC Context. The `AuthHandler` is retrieved from the `AuthInterceptor` instance present in the current gRPC Context. 

There may be multiple `AuthInterceptor` instances registered in the system. One `AuthInterceptor` is registered per `AuthHandler` implementation found in the classpath. This happens in `io.pravega.controller.server.rpc.auth.registerInterceptors()` when the gRPC Server is constructed.  

At runtime, for each gRPC call, the `Principal` and `AuthInterceptor` in turn are set by one of the multiple `AuthInterceptors`  that might be registered in the system. The one that will do it is the one that has an `Authhandler` that matches the auth method (`Bearer`, `Basic`, etc.) specified by the client. Please see the code below: 
https://github.com/pravega/pravega/blob/7a923ed9dac4688ff1eed13a11d876292f6f90c2/controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthInterceptor.java#L59

Hope this clarifies. 

Also, I had added a new test to verify a scenario involving multiple AuthHandler plugins. It is here: https://github.com/pravega/pravega/blob/7a923ed9dac4688ff1eed13a11d876292f6f90c2/test/integration/src/test/java/io/pravega/test/integration/ControllerGrpcListStreamsTest.java#L127